types/result/result.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package result contains the Of result type, which is
+// either a value or an error.
+package result
+
+// Of is either a T value or an error.
+//
+// Think of it like Rust or Swift's result types.
+// It's named "Of" because the fully qualified name
+// for callers reads result.Of[T].
+type Of[T any] struct {
+	v   T // valid if Err is nil; invalid if Err is non-nil
+	Err error
```

Comment:
Why is this exported?

Reply:
Seemed easier than an accessor? It's a struct value type anyway.

The v field is unexported to make people check for errors or panic. That's not needed for Err.

🤷‍♂️

Reply:
I'd personally pick consistency here

Reply:
Updated. Unexported.